draft-ietf-lake-authz.md

```diff
@@ -632,6 +641,189 @@ If present, V decrypts and verifies opaque_state as received from W. If that ver
 with U is aborted.
 If the voucher response is successfully received from W, then V responds to U with EDHOC message_2 as described in {{V_2}}.
 
+## Reverse use with U as Responder {#reverse-u-responder}
+
+This section presents a protocol variant in which U is the EDHOC Responder.
+This may allow optimizations in certain constrained network technologies.
+For example, one use case is having V broadcast message_1, to which U responds with a message_2 whose EAD_2 field contains Voucher_Info.
+
+This is different from the EDHOC reverse message flow defined in {{Appendix A.2.2 of RFC9528}}, since we make no assumption about whether U or V is a CoAP server.
+
+### U is the Initiator {#u-initiator}
+
+For clarity, we first present the default scenario with U as Initiator, as described in {{protocol-overview}} and {{U-V}}.
+Note that Voucher_Info and Voucher are carried in EDHOC message_1 and message_2, respectively.
+
+~~~~~~~~~~~ aasvg
++-----+-----+                  +-----------+
+|     U     |                  |     V     |
+| Initiator |                  | Responder |
++-----+-----+                  +-----+-----+
+      |                              |
+      |                              |
+      |       EDHOC message_1        |                              W
+      +----------------------------->|                              |
+      | EAD_1 = (TBD1, Voucher_info) |                              |
+      |                              |             VREQ             |
+      |                              +----------------------------->|
+      |                              | (SS, G_X, Voucher_Info,      |
+      |                              |  H_message_1, ?opaque_state) |
+      |                              |                              |
+      |                              |             VRES             |
+      |                              +<---------------------------->|
+      |                              |    (Voucher, ?opaque_state)
+      |       EDHOC message_2        |
+      +<-----------------------------|
+      |   EAD_2 = (TBD2, Voucher)    |
+      |                              |
+      |       EDHOC message_3        |
+      +----------------------------->|
+      |                              |
+      |                              |
+~~~~~~~~~~~
+{: #fig-u-initiator title="ELA when U is the EDHOC Initiator." artwork-align="center"}
+
+### U is the Responder {#u-responder}
+
+ELA also works with U as the EDHOC Responder, a setup we refer to as the ELA reverse flow, as shown in {{fig-u-responder}}.
+
+We present this variant as a set of changes to the regular protocol flow, so that U can run as the EDHOC Responder.
+The changes include:
+
+* Voucher_Info and Voucher are transported in EDHOC message_2 and message_3, respectively (instead of message_1 and message_2).
+* The EAD_2 and EAD_3 fields are identified with labels TBD1 and TBD2, respectively.
```

Comment:
EAD_2 and EAD_3 are generic EDHOC fields. They are not identified with labels, they *carry* labels TBD1 and TBD2 in this protocol variant to denote Vocuher_Info and Voucher are within the field.

Reply:
Thanks, fixed.